### PR TITLE
Fix setup benchmark

### PR DESCRIPTION
**Pull request by asi1024**
Enhanced and fixed `setup_benchmark`

### CODE REVIEW

File: benchmarks/functions/__init__.py [hunk 1]
```diff
@@ -27,11 +37,17 @@ def setup_benchmark(self, function, inputs, grad_outputs=None):
...
-        self.forward_inputs = (
-            [None if x is None else chainer.Variable(x) for x in inputs])
+
+        self.forward_inputs = ([self._convert_to_variable(x) for x in inputs])
 
         # Prepare for backward.
-        outputs = chainer.functions.identity(self.forward())
+        ret = self.forward()
+
+        if isinstance(ret, tuple):
```
Comment (kmaehashi) [inline benchmarks/functions/__init__.py]: How about this for simplicity?

```py
if not isinstance(ret, (list, tuple)):
    ret = (ret,)
outputs = chainer.functions.identity(*ret)
```

Comment (asi1024) [reply benchmarks/functions/__init__.py]: Simplified these lines by using _normalize_outputs().

File: benchmarks/functions/__init__.py [hunk 2]
```diff
@@ -18,6 +18,30 @@ class FunctionBenchmark(BenchmarkBase):
...
+            return None
+        elif isinstance(x, (int, float, bool, str)):
+            return x
+        elif isinstance(x, (list, tuple)):
+            return ([self._convert_to_variable(i) for i in x])
+        else:
+            return chainer.Variable(x)
+
+
+    # Make the format of outputs tuple of xp.ndarray/chainer.Variable.
```
Comment (kmaehashi) [inline benchmarks/functions/__init__.py]: * How about saying `Flatten outputs into tuple.`?
  It's much easier to understand.
* Let's use docstring.

File: benchmarks/functions/__init__.py [hunk 3]
```diff
@@ -18,6 +18,30 @@ class FunctionBenchmark(BenchmarkBase):
...
+    def _normalize_outputs(self, x):
+        def rec(x):
+            if isinstance(x, (list, tuple)):
+                ret = []
```
Comment (kmaehashi) [inline benchmarks/functions/__init__.py]: Use variable name other than `i` as it sounds like an index. (e.g., `elem`)

File: benchmarks/functions/__init__.py (hunk 2 above)
Comment (asi1024) [reply benchmarks/functions/__init__.py]: Fixed.

File: benchmarks/functions/__init__.py (hunk 3 above)
Comment (asi1024) [reply benchmarks/functions/__init__.py]: Fixed.

**Comment by kmaehashi** [comment]
LGTM!

**Pull request merged**